www/src/utils/reach-router-state-manager.js

```diff
@@ -1,33 +1,58 @@
-import React from "react"
+import React, { Component, Fragment } from "react"
 import queryString from "query-string"
 import { navigate } from "@reach/router"
 
-// manage your state entirely within the router, so that it's copiable
-// https://gist.github.com/sw-yx/efd9ee71669413bca6a895d87e30742f
+const emptySearchState = { s: ``, c: [], d: [], v: [], sort: `recent` }
+class RRSM extends Component {
+  state = emptySearchState
 
-export default defaultURLState => Component => props => {
-  const { location } = props
-  const urlState = { ...defaultURLState, ...queryString.parse(location.search) }
-  const setURLState = newState => {
-    const finalState = { ...urlState, ...newState } // merge with existing urlstate
+  static defaultProps = {
+    defaultSearchState: {},
+  }
+
+  setUrlState = newState => {
+    const finalState = { ...this.state, ...newState }
+    // update RSSM state
+    this.setState({ ...finalState })
+
+    // sync url to RSSM
     Object.keys(finalState).forEach(function(k) {
```

Comment:
Maybe use reduce here? Something like:

```jsx
// note: there may be more complexity here
// but it seems easier to handle the truthy case and include, rather than
// falsy case(s) and exclude
const params = Object.keys(finalState).reduce((merged, key) => {
  if (finalState[key]) {
    merged[key] = finalState[key]
  }
  return merged
}, {})

return navigate(`${location.pathname}?${queryString.stringify(params)}`)
```

Reply:
I like it, done. The previous check eliminated teh "sort" param, so I added that, so it'll be consistent with what was before. If we decide to make sort robust later, we can take out / tweak it then.